contents/blog/sql-for-analytics.md

```diff
@@ -0,0 +1,133 @@
+
+For example, a basic statement to select the `$current_url` of pageview events that happened after June 1, 2023, and order by their count looks like this:
+
+```sql
+SELECT
+	properties.$current_url,
+	count()
+FROM
+	events
+WHERE timestamp > (2023-06-01) and properties.$current_url IS NOT NULL
+GROUP BY properties.$current_url
+ORDER BY count() desc
+```
+
+`count()` in query is an example of an SQL function. There are hundreds of these functions to use. They generally help you:
+
+1. Aggregate values like `avg()`, `sum()`, `max()`
+2. Modify or calculate values like `extract()`, `dateDiff()`, `multiIf()`
+
+To learn more about SQL queries, commands, and useful functions, check out our [SQL insights](/docs/product-analytics/sql) and [expressions](/docs/hogql/expressions) documentation.
+
+> **Where are queries written?** Queries are generally written in a UI connected to the database. This could be a CLI for a framework, a UI in a database service, or a visualization tool. Examples include PostHog, Metabase, MySQL WorkBench, and psql.
+
+### 3. Result
+
+As an output from the query on your data, you get a result. The result takes the form of basically a list of values you selected, ordered, grouped, and modified in the way you want. For example, the query from the last section returns a list like this:
+
+```json
+[
+	{
+		"properties.$current_url": "https://app.posthog.com/home",
+		"count()": 300
+	},
+	{
+		"properties.$current_url": "https://app.posthog.com/events",
+		"count()": 200
+	},
+	{
+		"properties.$current_url": "https://app.posthog.com/insights/new",
+		"count()": 100
+	},
+	//...more values
+]
+```
+
+Most analytics platforms transform the data from the result into a visual such as a table, formatted page, or graph. This enables further analysis.
+
+![SQL insight](../images/blog/sql-for-analytics/sql.png)
+
+## Why use SQL for analytics?
+
+There are two main reasons to use SQL for analytics: **customizability** and **availability**
+
+First, SQL enables you to retrieve and format data in any way you like. You aren’t constrained by the tools and visualizations an analytics platform provides to you. This enables you to fully customize your query and results to your goals.
+
+The options analytics platform provide are a subset of all possible queries since they likely use SQL to calculate and power those options. Using SQL yourself provides access to all possible SQL queries, data, and formatting. 
+
+![All possible SQL queries](../images/blog/sql-for-analytics/possible.png)
+
+Although analytics platforms try to cover as many types of analysis and data as possible, they can’t do everything. Many types of queries aren’t possible to template. This is because the format of data for everyone’s analytics is different. For example, the your filters might be too complex or your data might be in a format visualizations can’t handle. With SQL knowledge, you can create queries using your own data format and get results relevant to you.
```

Comment:
```suggestion
Although analytics platforms try to cover as many types of analysis and data as possible, they can’t do everything. Many types of queries aren’t possible to template. This is because the format of data for everyone’s analytics is different. For example, your filters might be too complex or your data might be in a format visualizations can’t handle. With SQL knowledge, you can create queries using your own data format and get results relevant to you.
```